### DISCUSSION TITLE
Integrating wallet adapter to an existing Wallet

### DISCUSSION THREAD

**Discussion by Ovodo** [aptos-labs/aptos-developer-discussions#130] [Questions] [answered]
Please I would love if someone help me clarify things. 
So i already have an existing wallet sort of on the chrome extension store with functionalites such as creating a new account, and checking account balance. I want to add the functionality to be able to connect to dapps and also sign transactions from them. I have read the docs on the wallet adapter template plugin. They say to clone the repo and change some variables and then upload as a npm package. 
But i still dont get how this connects back to my wallet on the chrome webstore.
Can some one help me clarify things and how i should go about this properly.

Thanks.

**Comment by 0xmaayan** [comment, marked as answer]
Hi, if you are developing a wallet and want to integrate with our Aptos Wallet Adapter, best would be to be compatible with [AIP-62](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-62.md) as the legacy standard would soon be deprecated. You can follow this [guide](https://aptos.dev/integration/wallet-adapter-for-wallets#how-to-integrate-with-the-aip-62-wallet-standard) on how to integrate with the wallet adapter and check out [this wallet integration example](https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/apps/nextjs-example/utils/standardWallet.ts).

The wallet adapter serves as a communication tool between a dapp and a wallet to ease the dapp and wallet integration. You should implement the wallet functionality on the wallet side, but can be compatible with the wallet adapter interface for easy detection on the dapp side.

**Reply by Ovodo** [reply]
Thank you for this. I have gone through the documents but i am still a llittle bit confused. I built my current wallet as a browser extension with the sveltekit and implemented some of the functionalities manually already. What i want to know is that, is implementing the AptosWallet and AptosWalletAccount interfaces enough for me. Like can i just do that in one of the pages in my svelte app?.

And also i still dont seem to get how the npm package all links back to my browser extension wallet. like how will the dapp know that it is my particular wallet it is to connect to. how do i link my wallet with the code in the npm package. Also do i need to create a UI for signing transactions or this is done automatically.

And most importantly, how can i test all these before deploying to production. 
Thank you and i hope i am being understood a bit.

**Reply by 0xmaayan** [reply]
There are some parts here.

> What i want to know is that, is implementing the AptosWallet and AptosWalletAccount interfaces enough for me. Like can i just do that in one of the pages in my svelte app?

You need to inject your wallet into the browser window object to have it visible in the browser context

> And also i still dont seem to get how the npm package all links back to my browser extension wallet. like how will the dapp know that it is my particular wallet it is to connect to

You dont need a npm package, you need to register your wallet to let the dapp know the wallet is ready to be used https://aptos.dev/integration/wallet-adapter-for-wallets#register-wallet
From there, the adapter knows how to handle the communication

>  Also do i need to create a UI for signing transactions or this is done automatically

This needs to be done in the wallet

> And most importantly, how can i test all these before deploying to production.

You can use our demo dapp that supports the AIP-62 wallet standard https://aptos-labs.github.io/aptos-wallet-adapter/

**Reply by Ovodo** [reply]
![image](https://github.com/aptos-labs/aptos-developer-discussions/assets/107152769/18cf9bb4-5be7-48aa-a557-1bc7d49c1895)


Please for the url string here, am I supposed to put my chrome extension wallet url or website url.....

**Reply by 0xmaayan** [reply]
You should put the wallet website url where users can get more info and download the wallet from.

**Reply by Ovodo** [reply]
![image](https://github.com/aptos-labs/aptos-developer-discussions/assets/107152769/02b1d415-57fb-455d-afba-0f29fab6e8fa)

I am getting this error when trying to configure the plugin adapter. I followed the instructions for the repo and still get this. i think there is a bug somewhere.

**Reply by 0xmaayan** [reply]
Please make sure you use the latest `@aptos-labs/wallet-adapter-react` version with latest `@aptos-labs/ts-sdk` package version and have implemented the wallet standard interface 

https://github.com/aptos-labs/wallet-standard/tree/main/src/features

https://aptos.dev/integration/wallet-adapter-for-wallets#aip-62-wallet-standard

**Reply by Ovodo** [reply]
I think you are not getting me right. I am talking of the wallet adapter plugin.  I am not using any of the packages above you mentioned for the configuration.

This is the repo i am talking about.
https://github.com/aptos-labs/wallet-adapter-plugin-template.

Thank you.

**Reply by 0xmaayan** [reply]
I understand what you are trying to do, but if you look at previous messages my recommendation was to have your wallet support the AIP-62 new wallet standard, which means you dont need to implement this wallet plugin.

The only thing you need to do is to follow this doc https://aptos.dev/integration/wallet-adapter-for-wallets/#how-to-integrate-with-the-aip-62-wallet-standard

**Comment by Ovodo** [comment]
Thanks I think it's more clearer now.....i will try this and give you
feedback.....Thanks so much for your time and swift responses.

**Reply by Ovodo** [reply]
Here I am building my wallet extension with sveltekit. i have copied the exact wallet.ts file in the example to try it out. i have changed the wallet name and icon. I am currently testing this by loading an unpacked build in my browser extensions pages.
But i dont see the wallet show when i try it out with the demo dapp you posted earlier. 
Please where am i missing it. I need to be able to test the wallet properly so i can desing its interface well.
![image](https://github.com/aptos-labs/aptos-developer-discussions/assets/107152769/399c4c3e-153d-4310-97af-2b55a47067a5)

Thank you

**Reply by 0xmaayan** [reply]
have you imported `wallet.ts` file in the main framework file? Not familiar with sveltekit, but in react apps it is index.tsx file

**Reply by Ovodo** [reply]
No i thought i was not supposed to import the file. it doesnt export anything but the aptoswallet class and aptoswalletaccount classes. or should i just call the register function when the page loads. so if it were to be a react app, this will be calling the register function say on a use effect in the index.ts file. 
Please can i chat you up on discord or whatever medium u prefer so i can discuss it better.

**Reply by 0xmaayan** [reply]
you dont need to explicitly call any function but just `import ./wallet.ts` on the main file in your project so it would get bundled and used when you run your app.

**Reply by Ovodo** [reply]
its the same thing. I still cant see the wallet when i try the demo app.
I have imported it in the main page.svelte file.
![image](https://github.com/aptos-labs/aptos-developer-discussions/assets/107152769/958a8361-77a4-4e22-b767-638dab1a3273)

**Reply by 0xmaayan** [reply]
It is really hard to help with some small code pieces - could you share a repo?

**Reply by Ovodo** [reply]
Okay, here it is. It is the updrages branch
https://github.com/NetSepio/ChromiumExtension/tree/upgrades

**Reply by Ovodo** [reply]
Please Any Feedback on this?

**Reply by Ovodo** [reply]
Hello mayaan. please any feedback?.
I am importing the wallet.ts file and i can see Wallet Registered in my console but wallet doesnt still show when i test with the example dapp

**Reply by 0xmaayan** [reply]
The dapp needs to handle communication by listening to the wallet emitted events and display it on the UI. 

I highly recommend reading the AIP implementation reference https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-62.md#reference-implementation - it describes what a wallet and a dapp need to do in order to communicate with each other.

This is how the wallet adapter core handles the wallet<>dapp communication https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-core/src/WalletCore.ts#L154
keep it in a local state https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-react/src/WalletProvider.tsx#L61
and then display it in the UI https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/apps/nextjs-example/components/WalletButtons.tsx#L17

**Comment by Ovodo** [comment]
And also pls is there an even or smth I can use to detect the incoming
transaction from the app to enable me developer the UI for the wallet
whenever the app requests it.
